Frameworks/UIKit/UIApplicationMain.mm

```diff
@@ -164,33 +162,33 @@ int UIApplicationMainInit(NSString* principalClassName,
         }
     }
 
-    if (principalClassName == nil) {
+    // If a principalClassName was specified, allocate the specified UIApplication class
+    if (!principalClassName) {
         principalClassName = [infoDict objectForKey:@"NSPrincipalClass"];
     }
 
-    if (principalClassName != nil) {
+    if (principalClassName) {
         char* pClassName = (char*)[principalClassName UTF8String];
         uiApplication = [objc_getClass(pClassName) new];
     } else {
         uiApplication = [UIApplication new];
     }
 
-    id<UIApplicationDelegate> delegateApp;
-
-    if (delegateClassName != nil) {
+    // If a delegateClassName was specified, allocate the specified UIApplicationDelegate and set it on the UIApplication instance.
+    id<UIApplicationDelegate> appDelegate;
+    if (delegateClassName) {
         char* pClassName = (char*)[delegateClassName UTF8String];
-        delegateApp = [objc_getClass(pClassName) new];
+        appDelegate = [objc_getClass(pClassName) new];
```

Comment:
nit: could use `NSClassFromString(delegateClassName)` now, instead of manually unboxing.